File: FlutterApi/Entities/Models/Categories.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;

namespace Entities.Models
{

    [Table("Categories")]
    public  class Categories
    {
        public Categories()
        {
            Products = new HashSet<Products>();
        }
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }
        public string CategoryName { get; set; }
        public string photoUrl { get; set; }
        public int TotalItem { get; set; }

        public virtual ICollection<Products> Products { get; set; }

    }
}

File: lib/models/Categories.dart
class Categories {
  int id;
  String categoryName;
  String photoUrl;
  int totalItem;

  Categories({this.id, this.categoryName, this.photoUrl, this.totalItem});

  Categories.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    categoryName = json['categoryName'];
    photoUrl = json['photoUrl'];
    totalItem = json['totalItem'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['id'] = this.id;
    data['categoryName'] = this.categoryName;
    data['photoUrl'] = this.photoUrl;
    data['totalItem'] = this.totalItem;
    return data;
  }
}
